cmd/kubectl-gadget/utils/trace.go

```diff
@@ -208,65 +213,213 @@ func GenericTraceCommand(
+// function parameter.
+// A trace obtained with this function must be deleted calling DeleteTrace.
+func CreateTrace(config *TraceConfig) (string, error) {
+	traceID := randomTraceID()
+
+	labelsSelector := map[string]string{}
+	if config.Labels != "" {
+		pairs := strings.Split(config.Labels, ",")
+		for _, pair := range pairs {
+			kv := strings.Split(pair, "=")
+			if len(kv) != 2 {
+				return "", errors.New("labels should be a comma-separated list of key-value pairs (key=value[,key=value,...])")
+			}
+			labelsSelector[kv[0]] = kv[1]
 		}
 	}
 
-	if params.Node != "" && !nodeFound {
-		contextLogger.Fatalf("Invalid filter: Node %q does not exist", params.Node)
+	trace := &gadgetv1alpha1.Trace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: config.GadgetName + "-",
+			Namespace:    "gadget",
+			Annotations: map[string]string{
+				GADGET_OPERATION: config.Operation,
+			},
+			Labels: map[string]string{
+				TRACE_ID: traceID,
+				/*
+				 * Add all this information here to be able to find the trace thanks
+				 * to them when calling getTraceListFromParameters().
+				 */
+				"GadgetName":    config.GadgetName,
+				"nodeName":      config.NodeName,
+				"namespace":     config.NamespaceName,
+				"podName":       config.PodName,
+				"containerName": config.ContainerName,
+				"outputMode":    config.TraceOutputMode,
+			},
+		},
+		Spec: gadgetv1alpha1.TraceSpec{
+			Node:   config.NodeName,
+			Gadget: config.GadgetName,
+			Filter: &gadgetv1alpha1.ContainerFilter{
+				Namespace:     config.NamespaceName,
+				Podname:       config.PodName,
+				ContainerName: config.ContainerName,
+				Labels:        labelsSelector,
+			},
+			RunMode:    "Manual",
+			OutputMode: config.TraceOutputMode,
+		},
 	}
 
+	err := createTraces(trace)
+	if err != nil {
+		return "", err
+	}
+
+	return traceID, nil
+}
+
+// getTraceListFromID returns an array of pointer to gadgetv1alpha1.Trace
+// corresponding to the given traceID.
+// If no trace corresponds to this ID, error is set.
+func getTraceListFromID(traceID string) (gadgetv1alpha1.TraceList, error) {
 	var listTracesOptions = metav1.ListOptions{
-		LabelSelector: fmt.Sprintf("trace-template-hash=%s", traceID),
+		LabelSelector: fmt.Sprintf("%s=%s", TRACE_ID, traceID),
 		FieldSelector: fields.Everything().String(),
 	}
 
-	// Wait until results are ready and fetch all results
+	traceRestClient, err := getRestClient()
+	if err != nil {
+		return gadgetv1alpha1.TraceList{}, err
+	}
+
+	var traces gadgetv1alpha1.TraceList
+
+	err = traceRestClient.
+		Get().
+		Namespace("gadget").
+		Resource("traces").
+		VersionedParams(&listTracesOptions, scheme.ParameterCodec).
+		Do(context.TODO()).
+		Into(&traces)
+	if err != nil {
+		return traces, fmt.Errorf("Error getting traces from traceID %q: %w", traceID, err)
+	}
+
+	if len(traces.Items) == 0 {
+		return traces, fmt.Errorf("No traces found for traceID %q!", traceID)
+	}
+
+	return traces, nil
```

Comment:
I think you can create a common helper to use with `getTraceListFromID` and `getTraceListFromParameters` to avoid duplicating that code. I think it could receive as parameter the label selector (a string).